### ISSUE TITLE
False positive for `F722`?

### ISSUE THREAD

**Issue by gbaian10** [astral-sh/ruff#18063]
### Summary

I use [jaxtyping](https://github.com/patrick-kidger/jaxtyping) package to do some type checking tasks.

```py
import numpy as np
from jaxtyping import Float32


# For jaxtyping, which is correct content, ruff did not report any errors.
def foo(x: Float32[np.ndarray, "1"]) -> None: ...


# For jaxtyping, which is correct content, ruff reported an `F722` error.
def bar(x: Float32[np.ndarray, "1 2"]) -> None: ...


# For jaxtyping, which is `incorrect` content, ruff did not report any errors.
def baz(x: Float32[np.ndarray, "1, 2"]) -> None: ...
```

In the three examples above, only the second one will report an `F722` error.
The only difference between them is the string content. I’m not sure if this is a false positive.

### Version

ruff 0.11.9
